thavalon-server/src/connections/game_handlers.rs

```diff
@@ -0,0 +1,278 @@
+//! Module for all game-related REST endpoint handlers. This module also handles
+//! all websocket related functions.
+
+use crate::database::{accounts, games::DatabaseGame};
+use crate::lobby::{Lobby, LobbyChannel, LobbyCommand, LobbyError, LobbyResponse};
+
+use serde::{Deserialize, Serialize};
+use tokio::sync::{
+    mpsc::{self, Sender},
+    oneshot,
+};
+use warp::{
+    reject::{self, Reject},
+    reply,
+    ws::{WebSocket, Ws},
+    Rejection, Reply,
+};
+
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex};
+
+/// Type used for a global GameCollection of all active games.
+pub type GameCollection = Arc<Mutex<HashMap<String, LobbyChannel>>>;
+
+/// Serializeable response for a new game. Contains the friend code to join the game.
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NewGameResponse {
+    friend_code: String,
+}
+
+/// Deserializeable request to join a specified game.
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct JoinGameRequest {
+    friend_code: String,
+    display_name: String,
+}
+
+/// Serializable response from the server to a player attempting to join a game
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct JoinGameResponse {
+    socket_url: String,
+}
+
+/// Rejection for when a player does not have a verified email address.
+#[derive(Debug)]
+pub struct UnverifiedEmailRejection;
+impl Reject for UnverifiedEmailRejection {}
+
+/// Rejection for when a player attempts to join a nonexistant game.
+#[derive(Debug)]
+pub struct NonexistentGameRejection;
+impl Reject for NonexistentGameRejection {}
+
+/// Creates a new game for the given player ID.
+///
+/// # Arguments
+///
+/// * `player_id` - The Player ID of the game creator.
+/// * `game_collection` - The global store of active games.
+///
+/// # Returns
+///
+/// * `NewGameResponse` on success.
+/// * `UnverifiedEmailRejection` if the player's email isn't verified.
+pub async fn create_game(
+    player_id: String,
+    game_collection: GameCollection,
+) -> Result<impl Reply, Rejection> {
+    log::info!("Attempting to create new game for player {}.", player_id);
+
+    // Confirm email is verified
+    // if !verify_email(&player_id).await {
```

Comment:
if commented out code is intended, can we add a comment saying so? Also, do we want any frontend checks so user is forced to verify email when clicking play?